### PR TITLE
[KOGITO-8351] - Introduce Camel Add-on Serverless Workflow Example

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-8351

Depends on:

- https://github.com/kiegroup/kogito-runtimes/pull/2689

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `main` branch!**

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [kogito-examples] mandrel</b>

- for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch 
  Please add comment: <b>Jenkins (re)run [kogito-examples] mandrel-lts</b>
 
</details>

### CODE REVIEW

**Review by cristianonicolai — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-camel-routes/src/main/resources/camel.sw.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          "functionRef": {
+            "refName": "callSoap",
+            "arguments": {
+              "body": "${ .number }"
+            }
+          }
+        }
+      ],
+      "stateDataFilter": {
+        "output": "${ .words = .response[0] | del(.response) }"
```
Comment (fjtirado) [inline serverless-workflow-examples/serverless-workflow-camel-routes/src/main/resources/camel.sw.json]: The del(.response) part is not needed
You should write 
${ words : .response[0]}
which will set the model to {"words": the first item of the camel response}

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          "functionRef": {
+            "refName": "callSoap",
+            "arguments": {
+              "body": "${ .number }"
+            }
+          }
+        }
+      ],
+      "stateDataFilter": {
+        "output": "${ .words = .response[0] | del(.response) }"
```
Comment (fjtirado) [inline serverless-workflow-examples/serverless-workflow-camel-routes/README.md]: Leftover ;)

**Review by fjtirado — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md (hunk 2 above)
Comment (ricardozanini) [inline serverless-workflow-examples/serverless-workflow-camel-routes/README.md]:
```suggestion
        "output": "${ words = .response[0]}"
```

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# Kogito Serverless Workflow Camel Integration
+
+This repository exemplifies a new add-on that can be used in [Kogito Serverless Workflow projects](https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/) to call custom [Camel Routes](https://camel.apache.org/) within the same application (JVM).
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/README.md]:
```suggestion
This repository exemplifies the camel add-on that can be used in [Kogito Serverless Workflow projects](https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/) to call custom [Camel Routes](https://camel.apache.org/) within the same application (JVM).
```

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      - setHeader:
+          name: operationName
+          constant: NumberToWords
+      - toD:
+          uri: cxf://{{com.dataaccess.webservicesserver.url}}?serviceClass=com.dataaccess.webservicesserver.NumberConversionSoapType&wsdlURL=/wsdl/numberconversion.wsdl
+```
+
+## Requirements
+
+From the Camel perspective, the only requirement is that the route either return a primitive or a standard Java Bean object that can be marshaled to JSON.
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/README.md]:
```suggestion
From the Camel perspective, the only requirement is that the route either return a primitive or a standard Java Bean object that can be marshalled to JSON.
```

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    }
+  }
+}
+```
+
+The value is usually a `jq` expression that queries the workflow state for the object that will be used as an argument. It's the Camel route responsibility to handle the JSON data. In the example above, the argument is a primitive type (`int`) to keep the example as simple as possible.
+
+## Use Cases
+
+So when to use this approach? Kogito Serverless Workflow essentially has three options for state processing and service/event orchestration:
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/README.md]:
```suggestion
So when to use this approach?  
Kogito Serverless Workflow essentially has three options for state processing and service/event orchestration:
```

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+1. **Event-driven via CloudEvents**. Ideal for an Event-Driven architecture where the services are ready to consume and produce events working in more reactive way.
+2. **Sync or Async REST services invocations via OpenAPI/Async API**. There are options also to directly call a REST service in the architecture or ecosystem. Either async and sync methods are supported depending on your requirements.
+3. **Internal Service execution or invocation**. Kogito is also a framework to build Java Workflow applications. One can use it to create custom Java services to run a process within the same instance.
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/README.md:103]: why not adding those use cases to the guides ?

File: serverless-workflow-examples/serverless-workflow-camel-routes/pom.xml [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    <kogito.bom.artifact-id>kogito-bom</kogito.bom.artifact-id>
+    <kogito.bom.version>2.0.0-SNAPSHOT</kogito.bom.version>
+    <maven.compiler.release>11</maven.compiler.release>
+    <version.compiler.plugin>3.8.1</version.compiler.plugin>
+    <version.failsafe.plugin>3.0.0-M7</version.failsafe.plugin>
+    <!-- Used by the WSDL plugin -->
+    <version.org.apache.cxf>3.5.4</version.org.apache.cxf>
+    <!-- See: https://camel.apache.org/camel-quarkus/2.14.x/index.html -->
+    <!-- Aligned with Quarkus. We don't use the Camel Quarkus platform BOM to avoid upgrade delays in our CI. Feel free to use the BOM in your projects, though -->
+    <version.io.quarkus.camel>2.14.0</version.io.quarkus.camel>
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/pom.xml]: same as in runtimes here: https://github.com/kiegroup/kogito-runtimes/pull/2689#discussion_r1052323072

File: serverless-workflow-examples/serverless-workflow-camel-routes/src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# External property read by the Camel Route
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/src/main/resources/application.properties:1]: Please set the container properties: https://github.com/kiegroup/kogito-examples/blob/stable/serverless-workflow-examples/serverless-workflow-error-quarkus/src/main/resources/application.properties#L5-L13

**Review by radtriste — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-camel-routes/README.md (hunk 6 above)
Comment (ricardozanini) [reply serverless-workflow-examples/serverless-workflow-camel-routes/README.md:103]: I'll open a different PR

File: serverless-workflow-examples/serverless-workflow-camel-routes/pom.xml [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    <quarkus.platform.version>2.15.0.Final</quarkus.platform.version>
+    <kogito.bom.group-id>org.kie.kogito</kogito.bom.group-id>
+    <kogito.bom.artifact-id>kogito-bom</kogito.bom.artifact-id>
+    <kogito.bom.version>2.0.0-SNAPSHOT</kogito.bom.version>
+    <maven.compiler.release>11</maven.compiler.release>
+    <version.compiler.plugin>3.8.1</version.compiler.plugin>
+    <version.failsafe.plugin>3.0.0-M7</version.failsafe.plugin>
+    <!-- Used by the WSDL plugin -->
+    <version.org.apache.cxf>3.5.4</version.org.apache.cxf>
+    <!-- See: https://camel.apache.org/camel-quarkus/2.14.x/index.html -->
```
Comment (radtriste) [inline serverless-workflow-examples/serverless-workflow-camel-routes/pom.xml]:
```suggestion
```

**Review by radtriste — APPROVED** [review]
minor comment

**Pull request merged**